File: coco/src/main/java/coco/ep/m7/finalized/Neg.java
package coco.ep.m7.finalized;

public class Neg extends Exp implements coco.ep.m7.Neg<Exp> {

    protected Exp inner;

    public Neg(Exp _inner) {
        this.inner = _inner;
    }

    public Exp getInner() {
        return inner;
    }
    
    public void setInner(coco.ep.Exp<Exp> _inner) {
        inner = convert(_inner);
    }
}

File: coco_csharp/CoCo/EP/M4/Neg.cs
using System.Collections.Generic;

namespace EP.M4 {
    public interface Neg<FT> : Exp<FT>, EP.M3.Neg<FT> {

        public void setInner(EP.Exp<FT> inner);
	
        void Exp<FT>.truncate (int level) {
            if (level > 1) {
                convert(getInner()).truncate(level-1);
            } else {
                setInner(lit(convert(getInner()).eval()));
            }
        }
	
        EP.Exp<FT> Exp<FT>.simplify() {
            if (convert(getInner()).eval() == (0.0)) {
                return lit(0.0);
            } else {
                return neg(convert(getInner()).simplify());
            }
        }

        List<double> Exp<FT>.collect() {
            List<double> result = new List<double>();
            result.AddRange(convert(getInner()).collect());
            return result;
        }
    }
}